File: packages/api-models-ts/src/SkymapTriggers.ts
/**
 * Request and response models for `/api/skymap_trigger`.
 *
 * @since 1.0.0
 */

import * as v from "valibot";

import * as Schemas from "./Schemas.ts";

/**
 * The skymap-based triggers currently queued on a remote facility.
 *
 * There is no upstream SQLAlchemy model: the names come straight back from the
 * instrument's remote observation plan API.
 *
 * @since 1.0.0
 * @category Models
 */
export const SkymapTriggerQueue = Schemas.model(
    v.strictObject({
        trigger_names: Schemas.list(v.string()),
    })
);

/**
 * @since 1.0.0
 * @category Models
 */
export type SkymapTriggerQueue = v.InferOutput<typeof SkymapTriggerQueue>;

File: packages/api-models-py/skyportal_py_models/skymap_triggers.py
"""Request and response models for ``/api/skymap_trigger``."""

from __future__ import annotations

from pydantic import BaseModel, ConfigDict, Field


class SkymapTriggerQueue(BaseModel):
    """The skymap-based triggers currently queued on a remote facility.

    There is no upstream SQLAlchemy model: the names come straight back
    from the instrument's remote observation plan API.
    """

    model_config = ConfigDict(extra="forbid")

    trigger_names: list[str] = Field(default_factory=list)
